app/javascript/components/subject/GenericAction.js

```diff
@@ -0,0 +1,235 @@
+      this.setState({
+        showMonitoringPlanModal: true,
+        message: `monitoring plan to "${event.target.value}"`,
+        message_warning: '',
+        monitoring_plan: event?.target?.value ? event.target.value : '',
+      });
+    } else if (event?.target?.id && event.target.id === 'public_health_action') {
+      if (!this.state.patient.monitoring) {
+        this.setState({
+          showPublicHealthActionModal: true,
+          message: `latest public health action to "${event.target.value}"`,
+          message_warning:
+            'Since this record is on the "Closed" line list, updating this value will not move this record to another line list. If this individual should be actively monitored, please update the record\'s Monitoring Status.',
+          public_health_action: event?.target?.value ? event.target.value : '',
+        });
+      } else if (this.state.patient.isolation) {
+        this.setState({
+          showPublicHealthActionModal: true,
+          message: `latest public health action to "${event.target.value}"`,
+          message_warning: 'This will not impact the line list on which this record appears.',
+          household_warning:
+            'If any household members are being monitored in the exposure workflow, those records will appear on the PUI line list if any public health action other than "None" is selected above. If any household members are being monitored in the isolation workflow, this update will not impact the line list on which those records appear.',
+          public_health_action: event?.target?.value ? event.target.value : '',
+        });
+      } else {
+        this.setState({
+          showPublicHealthActionModal: true,
+          message: `latest public health action to "${event.target.value}"`,
+          message_warning:
+            event.target.value === 'None'
+              ? 'The monitoree will be moved back into the primary status line lists.'
+              : 'The monitoree will be moved into the PUI line list.',
+          household_warning:
+            'If any household members are being monitored in the exposure workflow, those records will appear on the PUI line list if any public health action other than "None" is selected above. If any household members are being monitored in the isolation workflow, this update will not impact the line list on which those records appear.',
+          public_health_action: event?.target?.value ? event.target.value : '',
+        });
+      }
+    } else if (event?.target?.name && event.target.name === 'apply_to_group') {
+      let applyToGroup = event.target.id === 'apply_to_group_yes';
+      this.setState({ [event.target.name]: applyToGroup });
+    } else if (event?.target?.id) {
+      let value = event.target.type === 'checkbox' ? event.target.checked : event.target.value;
+      this.setState({ [event.target.id]: event?.target?.value ? value : '' });
```

Comment:
Shouldn't these existence-operators (`?`) be on line 83 above? Feels like this ternary/setState bit could be re-written slightly.

Reply:
done